### PR TITLE
IGAPP-492: Proposal

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/modules/layout/containers/HeaderContainer.js [hunk 1]
```diff
@@ -84,15 +68,15 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
...
+    !languages || languages.status !== 'ready' || !stateCityCode) {
     // Route does not exist yet. In this case it is not really defined whether we are peek or not because
     // we do not yet know the city of the route.
     return {
       language: state.contentLanguage,
       routeCityModel,
       peeking: false,
       categoriesAvailable: false,
-      shareUrl: null
+      shareUrl: simpleRouteShareUrl
```
Comment (klinzo) [inline native/src/modules/layout/containers/HeaderContainer.js:79]: Optional: Keep the `shareUrlFromScene` above and change this to
```suggestion
      shareUrl: shareUrlFromScene || simpleRouteShareUrl
```.
It is not really needed though as the only routes with share url still in the route params will be image and pdf routes which are not using the normal header and therefore this header container.

**Comment by klinzo** [comment]
@SteffiStuffel I thought about some further refactoring of your code regarding the share url. What do you think? Imo this makes it a little bit easier to understand and simpler code. You don't have to use this though. Maybe some comments would be helpful.

File: native/src/modules/layout/containers/HeaderContainer.js [hunk 2]
```diff
@@ -51,28 +47,16 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
...
-
-    return {
-      language: propsRoute.params?.languageCode,
-      routeCityModel: undefined,
-      peeking: false,
-      categoriesAvailable: false,
-      shareUrl: shareUrlFromScene || shareUrl
-    }
-  }
+  const simpleRoutes = [OFFERS_ROUTE, DISCLAIMER_ROUTE, SPRUNGBRETT_OFFER_ROUTE]
```
Comment (SteffiStuffel) [inline native/src/modules/layout/containers/HeaderContainer.js:50]: makes it really clearer to give them a name

File: native/src/modules/layout/containers/HeaderContainer.js [hunk 3]
```diff
@@ -110,17 +94,16 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
...
-  }
+  const path = route.path || undefined
+  const shareUrl = path
+    ? url(path)
+    : cityContentUrl({
+      cityCode: stateCityCode,
+      languageCode: route.language,
+      route: routeName,
+      path: route.type || null
+    })
```
Comment (SteffiStuffel) [inline native/src/modules/layout/containers/HeaderContainer.js:106]: This is a lot handier than my solution.

**Review by SteffiStuffel — COMMENTED** [review]
I tested this and it works perfectly, (unless i miss something). Merged it into my branch

**Pull request merged**